File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int result = nums[0];
        int count = 0;
        for (int num: nums){
            if (num == result){
                count++;
            }
            else{
                count--;
                if (count < 0){
                    result = num;
                    count = 0;   
                }
            }
        }
        return result;
    }
}

File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result = nums[0]
        count = 0
        for num in nums:
            if num == result:
                count += 1
            else:
                count -= 1
                if count < 0:
                    result = num
                    count = 0
        
        return result
        